packages/core/src/auto-view/auto-headers.tsx

```diff
@@ -0,0 +1,79 @@
+import {get} from 'json-pointer';
+
+import {objectSchemaAsArray, ObjectSchemaAsArrayRules} from '../utils';
+import {JSONPointer} from '../repository';
+
+import {getHints} from './utils';
+import {AutoViewProps} from './auto-view';
+
+export interface AutoHeadersProps extends AutoViewProps {
+    children: (props: AutoViewProps) => JSX.Element;
+    path?: JSONPointer;
+}
+
+const ensureObjectData = (data: any): Record<string, any> => {
+    if (typeof data !== 'object') {
+        return {};
+    }
+
+    if (Array.isArray(data)) {
+        return data.reduce((acc, item) => {
+            if (typeof item !== 'object' || Array.isArray(item)) {
+                return acc;
+            }
+
+            return {...acc, ...item};
+        }, {} as Record<string, any>);
+    }
+
+    return data;
+};
+
+export const AutoHeaders = (props: AutoHeadersProps) => {
```

Comment:
Also, keep in mind: adding a feature is a minor version, removing one is a major. If we even decide to move this to a separate package, it would be a breaking change.